corehq/apps/data_analytics/malt_generator.py

```diff
@@ -37,14 +38,32 @@ def generate_malt(monthspans, domains=None):
     """
     domain_names = domains or Domain.get_all_names()
     for domain_name in domain_names:
+        last_submission_date = get_last_form_submission_received(domain_name)
+        last_malt_run_dates_by_month = _get_last_run_date_for_malt_by_month(domain_name, monthspans)
         for monthspan in monthspans:
-            logger.info(f"Building MALT for {domain_name} for {monthspan}")
-            all_users = get_all_user_rows(domain_name, include_inactive=False, include_docs=True)
-            for users in chunked(all_users, 1000):
-                users_by_id = {user['id']: CouchUser.wrap_correctly(user['doc']) for user in users}
-                malt_row_dicts = _get_malt_row_dicts(domain_name, monthspan, users_by_id)
-                if malt_row_dicts:
-                    _save_malt_row_dicts_to_db(malt_row_dicts)
+            last_malt_run_date = last_malt_run_dates_by_month.get(monthspan.startdate.date(), monthspan.startdate)
+            if last_submission_date and last_submission_date >= last_malt_run_date:
```

Comment:
What happens to rows that do not need to be updated? Are they excluded from the report, or set to zero? Does that change existing behavior?

Reply:
Existing behavior is that MALTRows are **not** created if they do not have any app submission data. See https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/data_analytics/malt_generator.py#L46-L47 on master, granted that has some of my recent changes but I kept that check in place from pre-existing code.

Reply:
Related, if a MALTRow is not created for that domain, the next time the malt is generated for that month, it will run on this domain again since there isn't a last_run_date to reference. This isn't a huge deal since these domains have zero submissions for that month and are typically fast, and maybe worth it to avoid created useless entries in the MALTRow table that can be conveyed just as easily by not creating the entry at all.